src/hera/workflows/io.py

```diff
@@ -0,0 +1,102 @@
+"""Input/output models for the Hera runner. TODO move to hera.workflows.runner package."""
+from collections import ChainMap
+from typing import Any, Dict, List, Tuple, Type, Union
+
+from hera.shared._pydantic import BaseModel
+from hera.workflows.artifact import Artifact
+from hera.workflows.parameter import Parameter
+
+try:
+    from inspect import get_annotations  # type: ignore
+except ImportError:
+    from hera.workflows._inspect import get_annotations  # type: ignore
+
+try:
+    from typing import Annotated, get_args, get_origin  # type: ignore
+except ImportError:
+    from typing_extensions import Annotated, get_args, get_origin  # type: ignore
+
+
+def _is_annotated_as(annotation: Any, class_or_tuple: Union[Type, Tuple[Type, ...]]):
+    return get_origin(annotation) is Annotated and isinstance(get_args(annotation)[1], class_or_tuple)
+
+
+class RunnerInput(BaseModel):
+    """Input model."""
+
+    @classmethod
+    def _get_parameters(cls) -> List[Parameter]:
+        parameters = []
+        annotations = {k: v for k, v in ChainMap(*(get_annotations(c) for c in cls.__mro__)).items()}
+
+        for field in cls.__fields__:
+            if get_origin(annotations[field]) is Annotated:
+                if isinstance(get_args(annotations[field])[1], Parameter):
+                    param = get_args(annotations[field])[1]
+                    if cls.__fields__[field].default:
+                        param.default = cls.__fields__[field].default
```

Comment:
Good point, thought it would be serialized from the `root_validator` but that wouldn't happen if we're doing a late init. The problem doesn't exist for normal func inputs because we create a new Parameter with the default here https://github.com/argoproj-labs/hera/blob/7982f90c604d1c1c7c87ce120d6f7fbe10f36398/src/hera/workflows/script.py#L337